src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -135,12 +135,15 @@ function useChatUtilsHook(): UseChatUtilsHook {
     return null;
   };
 
-  const goBackFromChatScreen = () => {
-    navigation.goBack();
-    setChat({
-      ...chat,
-      selectedChannel: null
-    });
+  const goBackFromChatScreen = async () => {
+    await Keyboard.dismiss();
+    setTimeout(() => {
+      navigation.goBack();
+      setChat({
+        ...chat,
+        selectedChannel: null
+      });
+    }, 350);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made in the `goBackFromChatScreen` function are generally good. However, using `setTimeout` can lead to potential issues if the component unmounts before the timeout finishes. This could result in trying to update the state on an unmounted component, which is not recommended. To avoid this, you should clear the timeout when the component unmounts.

Here's how you can do it:

```diff
  const goBackFromChatScreen = async () => {
    await Keyboard.dismiss();
+   const timerId = setTimeout(() => {
      navigation.goBack();
      setChat({
        ...chat,
        selectedChannel: null
      });
-   }, 350);
+   }, 350);
+   
+   return () => clearTimeout(timerId);
  };
```
This way, the timeout will be cleared when the component unmounts, preventing any potential errors or warnings about updating state on an unmounted component.


<!-- This is an auto-generated comment by OSS CodeRabbit -->